designer/server/src/main/scala/pl/touk/nussknacker/ui/definition/component/ComponentService.scala

```diff
@@ -70,19 +70,27 @@ class DefaultComponentService(
 
   import cats.syntax.traverse._
 
-  override def getComponentsList(implicit user: LoggedUser): Future[List[ComponentListElement]] = {
-    for {
-      components <- processingTypeDataProvider.all.toList.flatTraverse { case (processingType, processingTypeData) =>
+  override def getComponentsList(skipUsages: Boolean)(implicit user: LoggedUser): Future[List[ComponentListElement]] = {
+    val componentsFuture = processingTypeDataProvider.all.toList.flatTraverse {
+      case (processingType, processingTypeData) =>
         extractComponentsFromProcessingType(processingTypeData, processingType)
-      }
+    }
+
+    componentsFuture.flatMap { components =>
       // TODO: We should firstly merge components and after that create DTOs (ComponentListElement). See TODO in ComponentsValidator
-      mergedComponents = mergeSameComponentsAcrossProcessingTypes(components)
-      userAccessibleComponentUsages <- getUserAccessibleComponentUsages
-      enrichedWithUsagesComponents = mergedComponents.map(c =>
-        c.copy(usageCount = userAccessibleComponentUsages.getOrElse(c.id, 0))
-      )
-      sortedComponents = enrichedWithUsagesComponents.sortBy(ComponentListElement.sortMethod)
-    } yield sortedComponents
+      val mergedComponents = mergeSameComponentsAcrossProcessingTypes(components)
+
+      if (!skipUsages) {
+        getUserAccessibleComponentUsages.map { userAccessibleComponentUsages =>
+          val enrichedWithUsagesComponents = mergedComponents.map { c =>
+            c.copy(usageCount = userAccessibleComponentUsages.getOrElse(c.id, 0))
+          }
+          enrichedWithUsagesComponents.sortBy(ComponentListElement.sortMethod)
+        }
+      } else {
+        Future.successful(mergedComponents)
```

Comment:
I'm not sure if you're intentionally not sorting the components in case of `skipUsages = true`

Reply:
good catch, it was an oversight